### PR TITLE
Introduce SearchSelectField to the field components

### PR DESCRIPTION
**Pull request by mohib0306**
#### Summary

- `SearchSelectField`: introduce `SearchSelectField` built on top of the `SearchSelectInput` and `FieldLabel` to the available field components

### CODE REVIEW

File: .changeset/nervous-moose-pump.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@commercetools-uikit/search-select-field': minor
```
Comment (mohib0306) [inline .changeset/nervous-moose-pump.md:2]: Marked as minor change (learned from the releases last few days). Is this right?

File: .changeset/config.json [hunk 2]
```diff
@@ -93,7 +93,8 @@
       "@commercetools-uikit/toggle-input",
       "@commercetools-uikit/tooltip",
       "@commercetools-uikit/utils",
-      "@commercetools-uikit/search-select-input"
+      "@commercetools-uikit/search-select-input",
+      "@commercetools-uikit/search-select-field"
```
Comment (mohib0306) [inline .changeset/config.json:97]: Also added to the linked changeset list for version bumping. Not sure if it should be done now or after its first release.

File: packages/components/fields/search-select-field/src/search-select-field.js [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+} from '@commercetools-uikit/utils';
+import Constraints from '@commercetools-uikit/constraints';
+import Spacings from '@commercetools-uikit/spacings';
+import FieldLabel from '@commercetools-uikit/field-label';
+import SearchSelectInput from '@commercetools-uikit/search-select-input';
+import FieldErrors from '@commercetools-uikit/field-errors';
+
+const hasErrors = (errors) => errors && Object.values(errors).some(Boolean);
+const searchSelectInputProps = Object.keys(SearchSelectInput.propTypes);
+const filterSerachSelectFieldProps = (props) => {
```
Comment (mohib0306) [inline packages/components/fields/search-select-field/src/search-select-field.js:16]: Just a helper function to filter out wrong (invalid) props. It checks if the prop is allowed by `SearchSelectInput` and if yes, then adds it to the `filteredProps` list (valid list of props).

File: packages/components/fields/search-select-field/src/search-select-field.spec.js [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+it('should call onChange when changing the value', async () => {
+  const onChange = jest.fn();
+  renderSearchSelectField({
+    onChange,
+  });
+  const input = screen.getByLabelText('SearchSelectField');
+  fireEvent.focus(input);
+  fireEvent.change(input, { target: { value: 'mango' } });
+  await waitFor(() => screen.getByText('Mango'));
```
Comment (tdeekens) [inline packages/components/fields/search-select-field/src/search-select-field.spec.js:117]: I think this could be `findByText`?

**Review by tdeekens — COMMENTED** [review]
Looks like somebody used a generator :)

File: packages/components/fields/search-select-field/src/search-select-field.js (hunk 3 above)
Comment (jonnybel) [inline packages/components/fields/search-select-field/src/search-select-field.js:16]:
```suggestion
const filterSearchSelectFieldProps = (props) => {
```

Comment (jonnybel) [reply packages/components/fields/search-select-field/src/search-select-field.js:16]: Could you add this as a comment on the code? 🙂

**Comment by jonnybel** [comment]
Upon looking at the Percy snapshots (and then on the storybook), seems like field isn't properly displaying the different states (disabled, readonly, error, warning) 🤔

**Comment by mohib0306** [comment]
I will have a loot at it now

**Review by jonnybel — APPROVED** [review]

**Review by tdeekens — APPROVED** [review]

**Comment by mohib0306** [comment]
UX review passed. Merging this

**Pull request merged**